applications/OptimizationApplication/custom_utilities/filtering/explicit_filter.cpp

```diff
@@ -297,23 +342,49 @@ ContainerExpression<TContainerType> ExplicitVertexMorphingFilter<TContainerType>
 }
 
 template<class TContainerType>
-ContainerExpression<TContainerType> ExplicitVertexMorphingFilter<TContainerType>::FilterField(const ContainerExpression<TContainerType>& rContainerExpression) const
+ContainerExpression<TContainerType> ExplicitFilter<TContainerType>::FilterField(const ContainerExpression<TContainerType>& rContainerExpression) const
+{
+    return GenericFilterField<ExplicitFilterHelperUtilities::IntegratedWeight>(rContainerExpression);
+}
+
+template<class TContainerType>
+ContainerExpression<TContainerType> ExplicitFilter<TContainerType>::FilterIntegratedField(const ContainerExpression<TContainerType>& rContainerExpression) const
 {
-    return GenericFilterField<ExplicitVertexMorphingFilterHelperUtilities::IntegratedWeight>(rContainerExpression);
+    return GenericFilterField<ExplicitFilterHelperUtilities::NonIntegratedWeight>(rContainerExpression);
 }
 
 template<class TContainerType>
-ContainerExpression<TContainerType> ExplicitVertexMorphingFilter<TContainerType>::FilterIntegratedField(const ContainerExpression<TContainerType>& rContainerExpression) const
+void ExplicitFilter<TContainerType>::GetIntegrationWeights(ContainerExpression<TContainerType>& rContainerExpression) const
 {
-    return GenericFilterField<ExplicitVertexMorphingFilterHelperUtilities::NonIntegratedWeight>(rContainerExpression);
+    KRATOS_ERROR_IF_NOT(rContainerExpression.HasExpression())
+        << "Uninitialized container expression given. "
+        << rContainerExpression;
```

Comment:
Following check is not required because you are setting the `Expression` anyways. It does not matter there exists an expression or not. It will be replaced by new `Expression`
```suggestion
```